src/main/java/com/bobocode/creditadvisoryapp/model/Advisor.java

```diff
@@ -0,0 +1,65 @@
+package com.bobocode.creditadvisoryapp.model;
+
+import com.bobocode.creditadvisoryapp.enums.AppStatus;
+import com.bobocode.creditadvisoryapp.enums.Role;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Entity
+@Table(name = "advisors")
+public class Advisor extends User {
+
+  @Enumerated(EnumType.STRING)
+  private Role role;
+
+  @OneToMany(mappedBy = "advisor", cascade = {CascadeType.PERSIST, CascadeType.REMOVE}, orphanRemoval = true)
+  @ToString.Exclude
+  @Setter(AccessLevel.PRIVATE)
+  private List<Application> apps = new ArrayList<>();
+
+  public void addApplication(Application app) {
+    apps.add(app);
+    app.setAdvisor(this);
```

Comment:
There is a `validateAssignment()` method, but you only use it on the service level, however the method `addApplication()` is public, meaning anyone can call it in any place in the app and assign the application to an advisor ignoring your validation.
I would do assignment validation here.

Reply:
or you can use the approach described in this comment https://github.com/EEM86/creditAdvisoryApp/pull/1#discussion_r995596834